### ISSUE TITLE
[Bug]: Network endowments not properly updating after delete permitted network

### ISSUE THREAD

**Issue by hesterbruikman** [MetaMask/metamask-extension#28993]
### Describe the bug

Possibly related to:
https://github.com/MetaMask/metamask-extension/issues/28953
https://github.com/MetaMask/metamask-extension/issues/28954
https://github.com/MetaMask/metamask-extension/issues/29938
https://github.com/MetaMask/MetaMask-planning/issues/3735

When deleting a permitted network and navigating to the dapp permissions, the list no longer accurately updates.

### Steps to reproduce

TBD

### Error messages or log output

```shell

```

### Detection stage

In production (default)

### Version

12.9.0

### Build type

Beta

### Browser

Chrome

### Operating system

MacOS

**Comment by vpintorico** [comment]
@hesterbruikman are we still considering this issue as a release blocker?

**Comment by hesterbruikman** [comment]
Should be fixed by https://github.com/MetaMask/metamask-extension/pull/29855

@seaona could you verify this please? This ticket links several associated issues

**Comment by Ivan-dev-RED** [comment]
Still actual
![Image](https://github.com/user-attachments/assets/96c2afbd-9b3e-4162-b03c-375f50944cab)

**Comment by masonreid1729** [comment]
Yes, there’s some kind of givеаway going on. I got some Ethereum there yesterday for different wallets. It’s unlikely they’ll take it back from me, but it’s a pity I didn’t have more wallets and couldn’t get more. But maybe it’s for the best.

<img width="369" alt="Image" src="https://github.com/user-attachments/assets/26a83277-c0ba-4b1b-a08e-e434c0c75c8b" />

<img width="373" alt="Image" src="https://github.com/user-attachments/assets/12c0c289-2cb4-4c24-8143-3b3f42417b9a" />
